### DISCUSSION TITLE
Parametric Bottle Cap Generator - Teddy Warner

### DISCUSSION THREAD

**Discussion by giscus** [Twarner491/theodore.net#6] [General]
# Parametric Bottle Cap Generator - Teddy Warner

Lost a lid or just want a more functional cap? Generate and print your own, compatible with any existing threads!

https://teddywarner.org/Projects/ParametricGenerator/

**Comment by edalger** [comment]
Thanks for this information. I have created a bottle cap with it.
My cap diameter is 37 mm. 
I'm having trouble with the "Grips" however. 50 grips do not print properly. Do you have a suggestion as to how to calculate the correct grip count?
Thanks 
Ed, Loveland colorado

**Reply by Twarner491** [reply]
Hey Ed,
Your error might be machine dependent here, or potentially in your slicing profile. Depending on your machines tolerances & nozzle size, increasing the depth of the grips via the "GripDepth" parameter may help here. If this depth is too low, your slicer profile may ignore the grips all together, yielding a smooth cap. It also might be worth a shot decreasing number of grips all together (however don't dip below 11), as wider grips should also help trigger the feature in your slicer.
I can't say I've come up with a way to calculate grip count for the generator, as I've only really perused the trial and error method with grip count.
 Maybe this calls for a generator update - Ill be sure to tag this thread once I get around to it.
Best of luck, and glad you've found a use case for the generator,
-Teddy

**Reply by edalger** [reply]
Thank you Teddy for the response. I'll try some of your ideas and let you know what happens.
Best regards,
Ed...

**Comment by edalger** [comment]
Teddy, I see you're using a Prusa printer. My printer is a Prusa MK3s+ and has been installed and working for a year or so.
The PrusaSlicer is presented with a grip count as setup in Fusion 360. 50 count sets 50 entries in the slicer, 1 for each grip part. I feel that this isn't correct as the grip parts should be a single part entry coming from Fusion 360.
When the part is brought into the Slicer, the part graphic shows grip parts neatly around the edges.

When the .3mf file (created by Fusion 360) is presented by the Slicer, a question pops-up asking if the various parts should be treated as a single part. I answer Yes. This question-and-answer produce "grip count entries" in the slicer. That is, 50 grips produces 50 entries for the grips, not as a single part. (Slicer produces entries: Body62, Body63 through Body66, then Body674 through Body678m then Body691 through Body698, etc. ) These are displayed below "Print Settings" on the right side of the Slicer GUI.

I think the Slicer thinks that the grip parts are not a part of the cap itself but separate entries, if that makes any sense.

Ed...

**Comment by edalger** [comment]
Teddy, this is the message from the Slicer:

"This file contains several objects positioned at multiple heights.
Instead of considering them as multiple objects, should 
the file be loaded as a single object having multiple parts?"

If I answer no, then the Grip parts and presented as separated from the cap itself. Answering yes, produces and nice GUI picture of the grips around the cap edge, but the Slicer can't handle the individual print parts (50 count).
Ed...

**Reply by Twarner491** [reply]
Hi Ed, Just spent a minute playing around with importing a generated cap .3mf into PrusaSlicer - Your selected style of model import (treating multiple bodies as one part) is the way to go. 

Due to the nature of the .3mf file format, individual closed geometric bodies will be exported individually from Fusion 360. This is usually a super desirable feature, as exporting individual bodies will decrease file-size, instead of saving as a mesh (i.e. STL). However due to the nature of grip generation by the generator, each individual grip will be saved as its own body - thus the 50+ bodies in your slicers browser. 

If your slicer cannot handle this large body count, it might be worth a shot exporting your generated cap as a different file format from Fusion. Exporting with a mesh file format should fix your issue here, however you may also opt to use the "Combine Bodies" tool in Fusion to group all generated parts into one exportable body. Note this second fix may break the generator, and it will need to be reset before another use.

Best of luck!
Teddy

**Comment by edalger** [comment]
Thanks, Teddy, for the help with this situation.
The problem was resolved by simply exporting the Fusion360 project as a .STL file. It took about a minute to complete the export.
I then set the .STL to open in the PrusaSlicer and then saved as a .3fm  and  G-code files.
To note here, the individual grip files are no-longer in the .3fm file but contained in the .STL entry.
The G-code printed just fine with the 50 grips around the cap. :) It's a beautiful thing.

We have a supply of nice bottles with the stupid locking caps. We can now store various printer parts and other items with your printer cap design.
Thanks for the help with this Teddy. What you've done with your Fusion generator is pure genius. 
Ed, Loveland Colorado

**Reply by Twarner491** [reply]
Glad I could Help! 
Would love to see the final product - my site doesn't support file inclusion via the comment system, however all of my socials attached to this site would be great places to share.
Best of luck with future cap generation :)
-Teddy

**Comment by edalger** [comment]
Teddy, I have a picture of my cap but I don't know how/where to post it? I've looked around but didn't find "socials".

**Reply by Twarner491** [reply]
Hey Ed,
Either my [Instagram - @teddymakesstuff](https://www.instagram.com/teddymakesstuff/) or [Email - twarner491@gmail.com](mailto:<Twarner491@gmail.com>) would be super, thanks. Sorry, I guess I should of specified - all of my socials can be found in my site footer, as well as below my profile image on my "About Me" page.
-Teddy

**Comment by markeyro** [comment]
Awesome parametric!  Worked great for me.  

Just a quick feature request :)  Would be nice to be able to have a thicker walled cap sides and top but independently dialed in.  Just an idea! But again, awesome, and helped me with a project!!
